syntax = "proto3";

// Request Interface
message MsgDeliveryRequest {
  repeated RequestElement request_elements = 1;
}

message RequestElement {
  int32 id = 1;
  string interface = 2;
  int32 money = 3;
}

// Response Interface
message MsgDeliveryResponse {
  int32 id = 1;
  repeated MessageRecieved recv = 2;
}

message MessageRecieved {
  string interface = 1;
  ResponseResult result = 2;
}

message ResponseResult {
  int32 balance = 1;
  string status = 2;
}

// Service Definition
service Branch {
  rpc MsgDelivery(MsgDeliveryRequest) returns (MsgDeliveryResponse) {}
}
